src/swupd/manifest.go

```diff
@@ -0,0 +1,117 @@
+package swupd
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// ManifestHeader contains metadata for the manifest
+type ManifestHeader struct {
+	Format      uint16
+	Version     uint32
+	Previous    uint32
+	FileCount   uint32
+	TimeStamp   uint64
+	ContentSize uint64
+	Includes    []string
+}
+
+// Manifest represents a bundle or list of bundles (MoM)
+type Manifest struct {
+	Name         string
+	Header       ManifestHeader
+	Files        []File
+	AddedCount   uint32
+	RemovedCount uint32
+	ChangedCount uint32
+}
+
+// MoM is a manifest of manifests with the same header information
+type MoM struct {
+	Header       ManifestHeader
+	SubManifests []Manifest
+}
+
+// Read manifest file header
+func readManifestFileHeader(fields []string, m *Manifest) error {
```

Comment:
Are you missing a loop here, to read all of the header, or is this badly named?
At the moment you ignore unknown headers, is that the intention?

Reply:
This is poorly named. I haven't gotten to the code that will call this in a loop over the header lines.

Reply:
Yes, the intention was to ignore unknown headers.